test/integration/smoke/test_deploy_virtio_scsi_vm.py

```diff
@@ -96,6 +96,7 @@ def setUpClass(cls):
         cls.zone = get_zone(cls.apiclient, testClient.getZoneForTests())
         cls.pod = get_pod(cls.apiclient, cls.zone.id)
         cls.services['mode'] = cls.zone.networktype
+        cls.cleanup = []
```

Comment:
cls.cleanup is defined AFTER the if statement which checks if the hypervisor is supported.  
if the hypervisor ISN't supported, the teardown still occurs but 
```cleanup_resources(cls.apiclient, cls.cleanup)```
throws an exception because cls.cleanup hadn't been defined. @borisstoyanov

Reply:
had another look at the code, I think the problem is in that 'if' hypervisor check, see at the end of it there's a return, which will end the execution of this method and that's why we don't get the cleanup defined. See the issue is that instead of return we should have skip(), which will 'cancel' the test execution as it is. let me help with that